yarn-project/circuit-types/src/tx/tx_receipt.ts

```diff
@@ -9,9 +10,11 @@ import { TxHash } from './tx_hash.js';
  */
 export enum TxStatus {
   DROPPED = 'dropped',
-  MINED = 'mined',
   PENDING = 'pending',
-  REVERTED = 'reverted',
+  SUCCESS = 'success',
+  APP_LOGIC_REVERTED = 'app_logic_reverted',
+  TEARDOWN_REVERTED = 'teardown_reverted',
+  BOTH_REVERTED = 'both_reverted',
```

Comment:
Feels like we need a better name than "both-reverted", but let's roll with it for now